nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkComputation.java

```diff
@@ -70,6 +73,9 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
             } else {
                 new WorkHolder(work).run();
                 workIds.add(work.getId());
+                KeyValueStore kvStore = Framework.getService(KeyValueService.class)
+                                                 .getKeyValueStore(StreamWorkManager.KV_NAME);
+                kvStore.put(work.getId() + StreamWorkManager.STATE, Work.State.RUNNING.toString(), 60 * 60L);
```

Comment:
if the kv store hang or raise an error we are in trouble.
we can remove the kv access on the processing if instead of storing a state we store the (queue, offset) tuple, the getState will retrieve this tuple and returns  org.nuxeo.lib.stream.log.LogAppender#waitFor(offset, queue, Duration.ZERO)